cmake/everest-generate.cmake

```diff
@@ -494,7 +507,97 @@ function (ev_add_py_module MODULE_NAME)
             add_subdirectory(${MODULE_PATH})
         endif()
     else()
-        message(WARNING "C++ module ${MODULE_NAME} does not exist at ${MODULE_PATH}")
+        message(WARNING "Python module ${MODULE_NAME} does not exist at ${MODULE_PATH}")
+        return()
+    endif()
+
+    # FIXME (aw): this will override EVEREST_MODULES, might not what we want
+    set_property(
+        GLOBAL
+        PROPERTY EVEREST_MODULES ${EVEREST_MODULES}
+    )
+endfunction()
+
+function (create_rust_cargo_toml RUST_MODULE_LIST)
+    if(NOT ${EVEREST_ENABLE_RS_SUPPORT})
+        return()
+    endif()
+
+    set(OUTPUT_FILE_PATH "${CMAKE_BINARY_DIR}/modules/Cargo.toml")
+
+    # Write the static parts
+    file(WRITE "${OUTPUT_FILE_PATH}" "[workspace]\nresolver = \"2\"\nmembers = [\n")
+
+    # Append each symlink path
+    foreach(SYMLINK ${RUST_MODULE_LIST})
+        file(APPEND "${OUTPUT_FILE_PATH}" "    \"${SYMLINK}\",\n")
+    endforeach()
+
+    # Close the list
+    file(APPEND "${OUTPUT_FILE_PATH}" "]\n\n")
+
+    file(APPEND "${OUTPUT_FILE_PATH}" "[workspace.dependencies]\neverestrs = { path = \"")
+    file(APPEND "${OUTPUT_FILE_PATH}" "${everest-framework_SOURCE_DIR}/everestrs/everestrs")
+    file(APPEND "${OUTPUT_FILE_PATH}" "\" }\n")
+endfunction()
+
+function (ev_add_rs_module MODULE_NAME)
+    set(EVEREST_MODULE_INSTALL_PREFIX "${CMAKE_INSTALL_LIBEXECDIR}/everest/modules")
+
+    set(MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/${MODULE_NAME}")
+    set(MODULE_BUILD_PATH "${CMAKE_CURRENT_BINARY_DIR}/${MODULE_NAME}")
+
+    if(IS_DIRECTORY ${MODULE_PATH})
+        if(NOT ${EVEREST_ENABLE_RS_SUPPORT})
+            message(STATUS "Excluding Rust module ${MODULE_NAME} because EVEREST_ENABLE_RS_SUPPORT=${EVEREST_ENABLE_RS_SUPPORT}")
+            return()
+        elseif("${MODULE_NAME}" IN_LIST EVEREST_EXCLUDE_MODULES)
+            message(STATUS "Excluding module ${MODULE_NAME}")
+            return()
+        elseif(EVEREST_INCLUDE_MODULES AND NOT ("${MODULE_NAME}" IN_LIST EVEREST_INCLUDE_MODULES))
+            message(STATUS "Excluding module ${MODULE_NAME}")
+            return()
+        else()
+            message(STATUS "Setting up Rust module ${MODULE_NAME}")
+
+            # Symlink the source directory into the build directory
+            execute_process(COMMAND rm -rf "${MODULE_BUILD_PATH}")
```

Comment:
That feels incredibly dangerous

Reply:
How about
`file(REMOVE_RECURSE [<files>...])`
https://cmake.org/cmake/help/latest/command/file.html#remove-recurse
> Changed in version 3.15: Empty input paths are ignored with a warning.

Reply:
But then again, the actual deletion is the critical part, I guess.

Reply:
It is not even needed to be recursive, it is just a symlink. Ideally I want the equivalent of `rm {file} || true`.

Reply:
> It is not even needed to be recursive, it is just a symlink. Ideally I want the equivalent of `rm {file} || true`.

Them `rm -f {file}` would be correct. You don't need to recurse on deleting a symlink, and no danger of clobbering actual directories.

Reply:
then you could use
```cmake
execute_process(COMMAND rm ${MODULE_BUILD_PATH}")
```

or even

```cmake
execute_process(COMMAND unlink ${MODULE_BUILD_PATH}")
```

errors in execute_process do not stop cmake from continuing (at least by default), you would have to specify a RESULT_VARIABLE and check that if you would want that

Reply:
I went with `rm -f`, this has the correct semantics: delete a single file (symlink) if it is there or do nothing if nothing is there.